tonic/src/transport/service/tls.rs

```diff
@@ -76,11 +79,10 @@ impl TlsConnector {
             .await?;
 
         let (_, session) = io.get_ref();
-        if session.alpn_protocol() != Some(ALPN_H2) {
-            return Err(TlsError::H2NotNegotiated)?;
+        match session.alpn_protocol() {
+            Some(ALPN_H2) | None if self.assume_http2 => Ok(BoxedIo::new(io)),
```

Comment:
I would break this out into an if statement I find that a bit easier to read.

Reply:
I got rid of the match and brought back if statements. See if you like it better.